wurst/file/FileIO.wurst

```diff
@@ -0,0 +1,109 @@
+package FileIO
+import public ChunkedString
+import AbilityIds
+import ErrorHandling
+
+/* 	FileIO for Patches 1.31 and up with chunked string support for big data.
+	Reads and writes using preload files by manipulating default ability tooltips.
+
+	Read a file:
+	>	let fileContent = new File("MyFileName.pld").readAsString()
+
+	Write a file:
+	>	new File("MyFileName.pld")..write("MyContent")..close()
+*/
+
+/** Abilities which tooltips will be used to load data */
+@configurable constant IO_ABILITIES = [AbilityIds.aerialShackles, AbilityIds.rocketAttack, AbilityIds.callToArms,
+						AbilityIds.calltoArms, AbilityIds.cloudofFog, AbilityIds.controlMagic,
+						AbilityIds.defend, AbilityIds.dispelMagic, AbilityIds.feedbackArcaneTower,
+						AbilityIds.feedback]
+
+public var CAN_READ_FILES = true
+
+public class File
+	var filename = ""
+
+	construct(string filename)
+		this.filename = filename
+
+	function write(string content)
+		let buffer = new ChunkedString()
+		buffer..append(content)..flush()
+		write(buffer)
+		destroy buffer
+
+	function write(ChunkedString buffer)
+		for i = 0 to buffer.chunkCount
+			let str = buffer.getChunk(i)
+			if validateInput(str) != null
```

Comment:
When there's a backslash or a quote you just log an error and plow on?

Reply:
errror( crashes threads my friend